e2e/testcafe-devextreme/tests/dataGrid/markup/T1240074_hoveringRows.ts

```diff
@@ -0,0 +1,27 @@
+import DataGrid from 'devextreme-testcafe-models/dataGrid';
+import { createWidget } from '../../../helpers/createWidget';
+import url from '../../../helpers/getPageUrl';
+import { getData } from '../helpers/generateDataSourceData';
+
+fixture.disablePageReloads`HoveringRows`.page(
+  url(__dirname, '../../container.html'),
+);
+
+test('Hover over a row in DataGrid', async (t) => {
+  const dataGrid = new DataGrid('#container');
+  const firstRow = dataGrid.getDataRow(0);
+
+  await t
+    .hover(firstRow.element)
+    .expect(firstRow.element.hasClass('dx-state-hover'))
```

Comment:
The ```DataGrid``` is POM (Page Object Model)
Here is [a short article about POM](https://medium.com/tech-tajawal/page-object-model-pom-design-pattern-f9588630800b)
Instead of writing direct selectors in our tests, we try to encapsulate them in our widget-related POMs :)
Our widget-related POMs are stored [in the testcafe-models nx project](https://github.com/DevExpress/DevExtreme/tree/24_2/packages/testcafe-models)
And sometimes we call this entities just the ```Test Models``` or ```Test Object Models```

The [dataGrid.getDataRow](https://github.com/DevExpress/DevExtreme/blob/24_2/packages/testcafe-models/dataGrid/index.ts#L130) method returns the [DataRow object](https://github.com/DevExpress/DevExtreme/blob/24_2/packages/testcafe-models/dataGrid/data/row.ts#L20)
The DataRow already contains functionality to check the row hover [here](https://github.com/DevExpress/DevExtreme/blob/24_2/packages/testcafe-models/dataGrid/data/row.ts#L46)

So, let's replace the direct selector usage with the ```DataRow.isHovered``` usage :)
E.g:
```typescript
  .expect(firstRow.isHovered)
  .ok();
```